lib/httpmock.js

```diff
@@ -173,31 +208,13 @@ var server = http.createServer(function(request, response) {
         response.writeHead(matchingRule.status, matchingRule.headers);
         if (matchingRule.response !== "") {
             if (matchingRule.encoding === "") {
-                var acceptEncoding = request.headers["accept-encoding"];
-                trace("Accept-Encoding header: " + acceptEncoding);
-                if (!acceptEncoding) {
-                    acceptEncoding = "";
-                }
-
-                if (acceptEncoding.match(/\bgzip\b/)) {
-                    response.writeHead(matchingRule.status, {
-                        "Content-Encoding": "gzip"
-                    });
-                    zlib.gzip(matchingRule.response, (err, buffer) => {
-                        if (err) throw err;
-                        response.end(buffer);
-                    });
-                } else if (acceptEncoding.match(/\bdeflate\b/)) {
-                    response.writeHead(matchingRule.status, {
-                        "Content-Encoding": "deflate"
-                    });
-                    zlib.deflate(matchingRule.response, (err, buffer) => {
-                        if (err) throw err;
-                        response.end(buffer);
-                    });
-                } else {
-                    response.end(matchingRule.response);
-                }
+                handleAcceptEncoding(
+                    response,
+                    { ...request.headers, ...matchingRule.headers },
```

Comment:
Don't get what we are doing here, why are we mixing the request headers with the response headers?